src/GLONASS_IFB_CORRECTION/glonass_IFB_correction.c

```diff
@@ -0,0 +1,412 @@
+    || glo_IFB_casted->fix_outage > GLO_IFB_MAX_FIX_OUTAGE ) {
+
+    return 0;
+  }
+
+  return 1;
+}
+
+extern void glo_IFB_free(void *glo_IFB)
+{
+  assert( glo_IFB_is_valid(glo_IFB) );
+
+  free(glo_IFB);
+}
+
+extern void glo_IFB_copy(const void *glo_IFB_src, void *glo_IFB_dst)
+{
+  glo_IFB_t *glo_IFB_src_casted = (glo_IFB_t *) glo_IFB_src;
+  glo_IFB_t *glo_IFB_dst_casted = (glo_IFB_t *) glo_IFB_dst;
+
+  assert( glo_IFB_is_valid(glo_IFB_src) );
+  assert( glo_IFB_is_valid(glo_IFB_dst) );
+
+  /* field-to-field copy */
+  *glo_IFB_dst_casted = *glo_IFB_src_casted;
+}
+
+extern int glo_IFB_is_enough_sats(const rtk_t *rtk)
+{
+  int n_fix_sats = 0;
+  int n_fix_glo  = 0;
+  int sat;
+
+  assert( rtk_is_valid(rtk) );
+
+  /* count number of satellites available for ambiguity resolution */
+  for (sat = 0; sat < MAXSAT; sat++) {
+
+    if ( !rtk->ssat[sat].vsat[0] ) continue;
+
+    if ( rtk->ssat[sat].fix[0] == 2
+      || rtk->ssat[sat].fix[0] == 3 ) { /* 'fix' of 'hold' satellite status */
+
+      n_fix_sats++;
+      if ( rtk->ssat[sat].sys == SYS_GLO ) n_fix_glo++;
+    }
+  }
+
+  if ( (n_fix_sats < GLO_IFB_MIN_SATS)
+    || (n_fix_glo < GLO_IFB_MIN_GLO_SATS) ) {
+
+    return 0;
+  }
+
+  return 1;
+}
+
+extern double glo_IFB_get_glo_dt(const void *glo_IFB)
+{
+  glo_IFB_t *glo_IFB_casted = (glo_IFB_t *) glo_IFB;
+
+  assert( glo_IFB_is_valid(glo_IFB) );
+
+  return glo_IFB_casted->glo_dt;
+}
+
+extern double glo_IFB_get_delta_glo_dt(const void *glo_IFB)
+{
+  glo_IFB_t *glo_IFB_casted = (glo_IFB_t *) glo_IFB;
+
+  assert( glo_IFB_is_valid(glo_IFB) );
+
+  return glo_IFB_casted->delta_glo_dt;
+}
+
+extern void glo_IFB_send_signal_to_reset(void *glo_IFB)
+{
+  glo_IFB_t *glo_IFB_casted = (glo_IFB_t *) glo_IFB;
+
+  assert( glo_IFB_is_valid(glo_IFB) );
+
+  glo_IFB_casted->signal_to_reset = GLO_IFB_SIGNAL_TO_RESET;
+}
+
+/* -------------------------------------------------------------------------------------------------------------- */
+/* timestep operations */
+
+static void search_mode_step(glo_IFB_t *glo_IFB, rtk_t *rtk)
+{
+  assert( glo_IFB_is_valid(glo_IFB) );
+  assert( glo_IFB->mode == GLO_IFB_SEARCH_MODE );
+
+  /* do not produce search step if not enough sats available */
+  if ( !glo_IFB_is_enough_sats(rtk) ) {
+
+    return;
+  }
+
+  glo_IFB->glo_dt += GLO_IFB_SEARCH_STEP;
+
+  if ( glo_IFB->glo_dt > GLO_IFB_MAX_GLO_DT_SEARCH ) {
+
+    glo_IFB->glo_dt -= GLO_IFB_MAX_GLO_DT_SEARCH - GLO_IFB_MIN_GLO_DT_SEARCH;
+  }
+}
+
+/* calculate delta glo_dt optimizing one-epoch glonass residuals */
+static double optimize_glo_dt_local(rtk_t *rtk)
+{
+  double delta_glo_dt, variance;
+  double design_matrix[NSATGLO];
+  int    freq_num[NSATGLO], freq_num_reference = 0;
+  double residuals[NSATGLO];
+  int    n_fix_glo = 0;
+  int    i, sat;
+  int    is_reference_defined = 0;
+
+  assert( rtk_is_valid(rtk) );
+
+  for (sat = 0; sat < MAXSAT; sat++) {
+
+    if ( rtk->ssat[sat].sys != SYS_GLO ) continue;
```

Comment:
Continue should be on a new line. This is hard to read.

Reply:
I'd also prefer a bool variable that has a name and explains why do we need to skip this cycle.